### PR TITLE
[Networking] #502 Add tll and prompt to requests

### LINKED ISSUE #502: [Relay] Add ttl and prompt to every request and response defined in specs

### PR DESCRIPTION
**Pull request by llbartekll**
# Description
- Add TTL and prompt properties to publish request
- Refactor protocol methods implementations

Resolves #502 

## How Has This Been Tested?
with integrations tests

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/Auth/AuthProtocolMethod.swift [hunk 1]
```diff
@@ -1,34 +1,28 @@
...
-        case .authRequest:
-            return 3001
-        case .pairingDelete:
-            return 1001
-        case .pairingPing:
-            return 1003
-        }
-    }
+struct AuthRequestProtocolMethod: ProtocolMethod {
+    var method: String = "wc_authRequest"
```
Comment (flypaper0) [inline Sources/Auth/AuthProtocolMethod.swift:5]: why var?

File: Sources/Auth/AuthProtocolMethod.swift [hunk 2]
```diff
@@ -1,34 +1,28 @@
...
+struct PairingPingProtocolMethod: ProtocolMethod {
+    var method: String = "wc_pairingPing"
+
+    var requestConfig = RelayConfig(tag: 1002, prompt: false, ttl: 30)
+
+    var responseConfig = RelayConfig(tag: 1003, prompt: false, ttl: 30)
+}
+
+
+struct PairingDeleteProtocolMethod: ProtocolMethod {
```
Comment (flypaper0) [inline Sources/Auth/AuthProtocolMethod.swift:22]: structs for ProtocolMethod looks better

File: Sources/WalletConnectNetworking/NetworkInteracting.swift [hunk 3]
```diff
@@ -8,11 +8,11 @@ public protocol NetworkInteracting {
     var socketConnectionStatusPublisher: AnyPublisher<SocketConnectionStatus, Never> { get }
     func subscribe(topic: String) async throws
     func unsubscribe(topic: String)
-    func request(_ request: RPCRequest, topic: String, tag: Int, envelopeType: Envelope.EnvelopeType) async throws
-    func requestNetworkAck(_ request: RPCRequest, topic: String, tag: Int) async throws
-    func respond(topic: String, response: RPCResponse, tag: Int, envelopeType: Envelope.EnvelopeType) async throws
-    func respondSuccess(topic: String, requestId: RPCID, tag: Int, envelopeType: Envelope.EnvelopeType) async throws
-    func respondError(topic: String, requestId: RPCID, tag: Int, reason: Reason, envelopeType: Envelope.EnvelopeType) async throws
+    func request(_ request: RPCRequest, topic: String, protocolMethod: ProtocolMethod, envelopeType: Envelope.EnvelopeType) async throws
```
Comment (flypaper0) [inline Sources/WalletConnectNetworking/NetworkInteracting.swift:11]: 👍

File: Sources/WalletConnectSign/Engine/Common/SessionEngine.swift [hunk 4]
```diff
@@ -59,28 +59,29 @@ final class SessionEngine {
         let chainRequest = SessionType.RequestParams.Request(method: request.method, params: request.params)
         let sessionRequestParams = SessionType.RequestParams(request: chainRequest, chainId: request.chainId)
 
-        let rpcRequest = RPCRequest(method: SignProtocolMethod.sessionRequest.method, params: sessionRequestParams)
-        try await networkingInteractor.request(rpcRequest, topic: request.topic, tag: SignProtocolMethod.sessionRequest.requestTag)
+        let rpcRequest = RPCRequest(method: SessionRequestProtocolMethod().method, params: sessionRequestParams)
```
Comment (flypaper0) [inline Sources/WalletConnectSign/Engine/Common/SessionEngine.swift:62]: SessionRequestProtocolMethod() to variable? To avoid typo later

Comment (flypaper0) [reply Sources/WalletConnectSign/Engine/Common/SessionEngine.swift:62]: but may be it better to isolate RPCRequest creation inside Networking package. But it another task of course

**Review by flypaper0 — APPROVED** [review]
lgtm, great job!

File: Sources/Auth/AuthProtocolMethod.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/Auth/AuthProtocolMethod.swift:5]: because xcode has generated it that as vars. Anyway good point, I am changing for lets

**Pull request merged**